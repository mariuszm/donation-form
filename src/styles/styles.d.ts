import 'styled-components';

import type { colors, componentsPalette } from './colors';
import type { fonts, typography } from './typography';

declare module 'styled-components' {
  export interface DefaultTheme {
    colors: typeof colors;
    fonts: typeof fonts;
    palettes: {
      component: typeof componentsPalette;
    };
    typography: typeof typography;
  }
}
