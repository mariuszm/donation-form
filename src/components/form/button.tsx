import { styled } from 'styled-components';

type Variant = 'primary' | 'white';

interface ButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  variant: Variant;
}

export const Button = styled.button<ButtonProps>`
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  height: 52px;
  padding: 16px 24px;
  border-radius: 5px;
  border: 1px solid ${({ theme }) => theme.colors.purpleGray};
  text-transform: capitalize;
  cursor: pointer;

  ${({ theme }) => theme.typography.form.button}

  background-color: ${({ theme, variant }) =>
    theme.palettes.component.button[variant].main};
  color: ${({ theme, variant }) =>
    theme.palettes.component.button[variant].text};

  &:hover {
    background-color: ${({ theme, variant }) =>
      theme.palettes.component.button[variant].hover};
  }

  &:active {
    background-color: ${({ theme, variant }) =>
      theme.palettes.component.button[variant].active};
  }
`;
