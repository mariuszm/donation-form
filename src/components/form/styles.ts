import CurrencyInput from 'react-currency-input-field';
import { styled } from 'styled-components';

export const FormWrapper = styled.div`
  width: 100vw;
  max-width: 600px;
  background-color: ${({ theme }) => theme.colors.white};
  box-shadow: 0 16px 32px 0 #1e2a3214;
  border-radius: 2px;
  overflow: hidden;
`;

export const Form = styled.form`
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(248px, 1fr));
  grid-column-gap: 24px;
  grid-row-gap: 32px;
  padding: 32px 40px;
`;

export const FormField = styled.div`
  display: flex;
  flex-direction: column;
  gap: 6px;
`;

export const FormCaption = styled.div`
  background-color: ${({ theme }) => theme.colors.salmon};
  height: 193px;
  padding-left: 40px;
  padding-right: 40px;
  position: relative;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  gap: 16px;

  div {
    display: flex;
    flex-direction: column;
    gap: 4px;
  }

  h1 {
    font-family: 'Work Sans';
    font-size: 24px;
    font-weight: 600;
    line-height: 120%;
    color: ${({ theme }) => theme.colors.midnightPurple};
  }

  p {
    font-family: 'Inter';
    font-size: 16px;
    font-weight: 400;
    line-height: 20px;
    color: ${({ theme }) => theme.colors.purpleGray};
  }
`;

export const ButtonClose = styled.div`
  width: 40px;
  height: 40px;
  position: absolute;
  top: 8px;
  right: 8px;
  cursor: pointer;
  background-color: ${({ theme }) => theme.palettes.component.closeButton.main};
  background-image: url('/svg/close-icon.svg');
  background-position: center center;
  background-repeat: no-repeat;
  border-radius: 10px;
  border: none;

  &:hover {
    background-color: ${({ theme }) =>
      theme.palettes.component.closeButton.hover};
  }
`;

export const InputAmount = styled(CurrencyInput)`
  ${({ theme }) => theme.typography.form.input}
  font-size: 24px;
  padding-left: 40px;
  padding-right: 16px;
  height: 60px;
  border-radius: 4px;
  border: 1px solid ${({ theme }) => theme.colors.gray};
  background-color: ${({ theme }) => theme.colors.white};
  background-image: url('/svg/dollar.svg');
  background-position: 8px center;
  background-repeat: no-repeat;
  color: ${({ theme }) => theme.colors.purpleGray};
`;

export const Total = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  gap: 24px;
  grid-column: 1 / -1;
  padding-top: 24px;
  border-radius: 4px;
  border: 1px solid ${({ theme }) => theme.colors.stroke};
`;

export const TotalInfo = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  gap: 8px;
  padding-left: 16px;
  padding-right: 16px;
`;

export const TotalLabel = styled.div`
  font-family: 'Work Sans';
  font-size: 16px;
  font-weight: 500;
  line-height: 120%;
  color: ${({ theme }) => theme.colors.blueGray900};
`;

export const TotalValue = styled.div`
  font-family: 'Inter';
  font-size: 24px;
  font-weight: 700;
  font-variant-numeric: tabular-nums;
  line-height: 120%;
  color: ${({ theme }) => theme.colors.purpleGray};
`;

export const InfoBox = styled.div`
  padding: 24px 16px;
  font-family: 'Work Sans';
  font-size: 12px;
  font-weight: 400;
  line-height: 140%;
  border-top-left-radius: 0;
  border-top-right-radius: 0;
  border-bottom-left-radius: 5px;
  border-bottom-right-radius: 5px;
  background-color: ${({ theme }) => theme.colors.stroke};
  color: ${({ theme }) => theme.colors.blueGray900};

  strong {
    font-family: 'Inter';
    font-weight: 700;
  }
`;

export const Label = styled.label`
  font-family: 'Work Sans';
  font-weight: 500;
  font-size: 14px;
  line-height: 18px;
  color: ${({ theme }) => theme.colors.midnightGray};
  cursor: text;
`;

export const Actions = styled.div`
  display: flex;
  grid-column: 1 / -1;
  gap: 27px;
  padding: 8px;
`;
