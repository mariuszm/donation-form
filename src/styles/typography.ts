import { colors } from './colors';

export const fonts = {
  inter: 'Inter',
  rubik: 'Rubik',
  work: 'Work Sans',
};

const defaults = {
  button: `
    font-family: ${fonts.work};
    font-size: 16px;
    font-weight: 600;
    line-height: 20px;
  `,
  label: `
    font-family: ${fonts.work};
    font-weight: 500;
    color: ${colors.midnightGray}
  `,
} as const;

export const typography = {
  form: {
    button: defaults.button,
    label: `
      ${defaults.label};
      font-size: 14px;
      line-height: 18px;
    `,
    input: `
      font-family: ${fonts.rubik};
      font-weight: 500;

      &::placeholder {
        color: #00000033;
      }
    `,
  },
} as const;
