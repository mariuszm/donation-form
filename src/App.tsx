import { useCallback, useState } from 'react';
import { type CurrencyInputProps } from 'react-currency-input-field';
import { formatValue } from 'react-currency-input-field';
import { styled } from 'styled-components';

import FormCaptionIcon from '@/assets/form-caption-icon.svg?react';
import Logo from '@/assets/naturally.svg?react';
import LogoIcon from '@/assets/spa.svg?react';
import { Button } from '@/components/form/button';
import {
  Actions,
  ButtonClose,
  Form,
  FormCaption,
  FormField,
  FormWrapper,
  InfoBox,
  InputAmount,
  Label,
  Total,
  TotalInfo,
  TotalLabel,
  TotalValue,
} from '@/components/form/styles';
import { MonthPicker } from '@/components/month-picker';
import { useDonationStore } from '@/store';

function App() {
  const [amount, setAmount] = useState<string | undefined>();
  const [formattedAmount, setFormattedAmount] = useState<string | undefined>();
  const [total, setTotal] = useState('0');
  const { dateUntil, getMonthName } = useDonationStore();

  const handleOnValueChange: CurrencyInputProps['onValueChange'] = (
    value,
    _,
    values,
  ) => {
    setAmount(value);
    setFormattedAmount(values?.formatted);
  };

  const handleOnChangeMonth = useCallback(
    (numberOfMonths: number) => {
      setTotal(
        formatValue({
          value: `${parseFloat(amount || '0') * numberOfMonths}`,
          decimalScale: 2,
        }),
      );
    },
    [amount],
  );

  return (
    <Container>
      <Header>
        <div>
          <LogoIcon />
          <Logo />
        </div>
      </Header>
      <FormWrapper>
        <FormCaption>
          <FormCaptionIcon />
          <div>
            <h1>The giving block</h1>
            <p>Set up your donation goal!</p>
          </div>
          <ButtonClose />
        </FormCaption>
        <Form>
          <FormField>
            <Label>I can donate</Label>
            <InputAmount
              value={amount}
              placeholder="0.00"
              onValueChange={handleOnValueChange}
              data-testid="input-amount"
            />
          </FormField>
          <FormField>
            <Label>Every month until</Label>
            <MonthPicker today={new Date()} onChange={handleOnChangeMonth} />
          </FormField>
          <Total>
            <TotalInfo>
              <TotalLabel>Total amount</TotalLabel>
              <TotalValue data-testid="total">${total}</TotalValue>
            </TotalInfo>
            <InfoBox data-testid="infobox">
              You will be sending <strong>${formattedAmount || '0.00'}</strong>{' '}
              every month, until{' '}
              <strong>
                {getMonthName()} {dateUntil?.getFullYear()}
              </strong>
              . Thank you!
            </InfoBox>
          </Total>
          <Actions>
            <Button variant="white">cancel</Button>
            <Button variant="primary">continue</Button>
          </Actions>
        </Form>
      </FormWrapper>
    </Container>
  );
}

const Header = styled.div`
  display: none;
  align-items: center;
  height: 80px;
  width: 100vw;
  padding-left: 40px;
  padding-right: 40px;
  background-color: ${({ theme }) => theme.colors.white};

  div {
    display: flex;
    gap: 8px;
    align-items: baseline;
  }
`;

const Container = styled.div`
  width: 100vw;
  height: 100vh;
  display: flex;
  align-items: center;
  flex-direction: column;
  gap: 64px;
  margin: auto;

  @media screen and (min-width: 600px) {
    ${Header} {
      display: flex;
    }

    ${FormCaption} {
      height: 128px;
      flex-direction: row;
      justify-content: flex-start;
      gap: 20px;

      h1 {
        font-size: 32px;
      }
    }

    ${Total} {
      border: none;
      padding-top: 0;
    }

    ${TotalLabel} {
      font-size: 20px;
    }

    ${TotalValue} {
      font-size: 32px;
    }

    ${InfoBox} {
      border-radius: 5px;
    }

    ${ButtonClose} {
      display: none;
    }
  }
`;

export default App;
