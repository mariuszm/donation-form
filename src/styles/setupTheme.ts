import type { DefaultTheme } from 'styled-components';

import { colors, componentsPalette } from './colors';
import { fonts, typography } from './typography';

export const theme: DefaultTheme = {
  colors,
  fonts,
  palettes: {
    component: componentsPalette,
  },
  typography,
};
