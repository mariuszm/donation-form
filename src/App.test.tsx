import { fireEvent, render, screen } from '@/test/test-utils';

import { InputAmount } from './components/form/styles';
import App from './App';

const setup = () => {
  const utils = render(<InputAmount />);
  const input = utils.container.firstChild as HTMLInputElement;
  return {
    input,
    ...utils,
  };
};

describe('App', () => {
  it('form is visible', () => {
    const { container } = render(<App />);
    expect(screen.getByText(/The giving block/i)).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('InputAmount', () => {
  it('is empty on load', () => {
    render(<App />);
    expect(screen.getByTestId('input-amount')).not.toHaveValue();
  });

  it('should allow numbers only', () => {
    const { input } = setup();
    expect(input.value).toBe('');

    fireEvent.change(input, { target: { value: 'abc' } });
    expect(input.value).toBe('');

    fireEvent.change(input, { target: { value: '123' } });
    expect(input.value).toBe('123');
  });

  it('should display value as money', () => {
    const { input } = setup();
    expect(input.value).toBe('');

    fireEvent.change(input, { target: { value: '12' } });
    expect(input.value).toBe('12');

    fireEvent.change(input, { target: { value: '12345' } });
    expect(input.value).toBe('12,345');

    fireEvent.change(input, { target: { value: '12345.55' } });
    expect(input.value).toBe('12,345.55');

    fireEvent.change(input, { target: { value: '0.5' } });
    expect(input.value).toBe('0.5');

    fireEvent.change(input, { target: { value: '.12345' } });
    expect(input.value).toBe('0.12');
  });
});
