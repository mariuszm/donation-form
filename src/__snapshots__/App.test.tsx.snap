// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > form is visible 1`] = `
<div
  class="sc-dJGMql eWKMMF"
>
  <div
    class="sc-ifyrAs ZVjB"
  >
    <div>
      <svg
        fill="none"
        height="32"
        viewBox="0 0 32 32"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          id="spa"
        >
          <mask
            height="32"
            id="mask0_12542_37"
            maskUnits="userSpaceOnUse"
            style="mask-type: alpha;"
            width="32"
            x="0"
            y="0"
          >
            <rect
              fill="#D9D9D9"
              height="32"
              id="Bounding box"
              width="32"
            />
          </mask>
          <g
            mask="url(#mask0_12542_37)"
          >
            <path
              d="M16.3998 30.2001C14.4887 30.0223 12.672 29.5334 10.9498 28.7334C9.22758 27.9334 7.70536 26.839 6.38314 25.4501C5.06092 24.0612 3.98869 22.4056 3.16647 20.4834C2.34425 18.5612 1.88869 16.389 1.7998 13.9667C4.04425 13.9667 6.04425 14.2834 7.7998 14.9167C9.55536 15.5501 11.0498 16.5279 12.2831 17.8501C13.5165 19.1723 14.4776 20.8556 15.1665 22.9001C15.8554 24.9445 16.2665 27.3779 16.3998 30.2001ZM15.9665 17.7667C15.2554 16.6112 14.4998 15.6723 13.6998 14.9501C12.8998 14.2279 11.8776 13.5667 10.6331 12.9667C10.7442 12.0779 10.9554 11.1223 11.2665 10.1001C11.5776 9.07786 11.9665 8.05564 12.4331 7.03342C12.8998 6.01119 13.4331 5.00008 14.0331 4.00008C14.6331 3.00008 15.2665 2.05564 15.9331 1.16675C16.6442 2.05564 17.3054 3.00008 17.9165 4.00008C18.5276 5.00008 19.072 6.02786 19.5498 7.08341C20.0276 8.13897 20.422 9.17786 20.7331 10.2001C21.0442 11.2223 21.2554 12.2001 21.3665 13.1334C20.3665 13.4445 19.3942 14.0223 18.4498 14.8667C17.5054 15.7112 16.6776 16.6779 15.9665 17.7667ZM19.3665 29.4001C19.2331 27.5779 19.0387 25.9334 18.7831 24.4667C18.5276 23.0001 18.1998 21.789 17.7998 20.8334C18.8665 18.6334 20.6165 16.8556 23.0498 15.5001C25.4831 14.1445 27.8665 13.6112 30.1998 13.9001C29.8442 18.0779 28.7498 21.4334 26.9165 23.9667C25.0831 26.5001 22.5665 28.3112 19.3665 29.4001Z"
              fill="#423C66"
              id="spa_2"
            />
          </g>
        </g>
      </svg>
      <svg
        fill="none"
        height="19"
        viewBox="0 0 107 19"
        width="107"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          id="Natur.ally"
        >
          <path
            d="M0 15.3125H2.8125V4.16667H4.04167L1.03125 2.26042L13.3958 15.3125H15.9896V0.28125H13.1875V11.3958H11.9479L14.9688 13.2917L2.59375 0.28125H0V15.3125Z"
            fill="#423C66"
          />
          <path
            d="M22.5938 15.5104C23.309 15.5104 23.9722 15.4271 24.5833 15.2604C25.1944 15.0868 25.7326 14.8438 26.1979 14.5313C26.6632 14.2118 27.0243 13.8368 27.2813 13.4063H27.3333V15.3125H30.1042V8.30208C30.1042 7.42014 29.8681 6.66319 29.3958 6.03125C28.9236 5.39931 28.2431 4.91319 27.3542 4.57292C26.4653 4.23264 25.3958 4.0625 24.1458 4.0625C23 4.0625 22.0069 4.22222 21.1667 4.54167C20.3333 4.85417 19.6667 5.28819 19.1667 5.84375C18.6736 6.39236 18.3715 7.02083 18.2604 7.72917L18.2396 7.85417H20.875L20.9063 7.76042C21.0451 7.25347 21.3854 6.85069 21.9271 6.55208C22.4757 6.25347 23.2014 6.10417 24.1042 6.10417C25.1389 6.10417 25.934 6.30556 26.4896 6.70833C27.0521 7.11111 27.3333 7.67708 27.3333 8.40625V10.8229C27.3333 11.3785 27.1632 11.8646 26.8229 12.2813C26.4896 12.691 26.0174 13.0104 25.4063 13.2396C24.8021 13.4618 24.0938 13.5729 23.2813 13.5729C22.441 13.5729 21.7639 13.441 21.25 13.1771C20.7431 12.9063 20.4896 12.5 20.4896 11.9583V11.9375C20.4896 11.5139 20.6701 11.1771 21.0313 10.9271C21.3993 10.6771 21.9896 10.5417 22.8021 10.5208L28.7292 10.375V8.71875L22.4167 8.84375C21.3194 8.87153 20.4236 9.02083 19.7292 9.29167C19.0417 9.5625 18.5382 9.93403 18.2188 10.4063C17.8993 10.8785 17.7396 11.4306 17.7396 12.0625V12.0729C17.7396 12.816 17.9375 13.4444 18.3333 13.9583C18.7361 14.4653 19.3021 14.8507 20.0313 15.1146C20.7604 15.3785 21.6146 15.5104 22.5938 15.5104Z"
            fill="#423C66"
          />
          <path
            d="M38.0833 15.5C38.4444 15.5 38.7882 15.4861 39.1146 15.4583C39.4479 15.4306 39.7465 15.3924 40.0104 15.3438V13.3646C39.8507 13.3924 39.6563 13.4167 39.4271 13.4375C39.2049 13.4583 38.9826 13.4688 38.7604 13.4688C37.9479 13.4688 37.3611 13.3229 37 13.0313C36.6458 12.7326 36.4688 12.2222 36.4688 11.5V6.33333H40.0104V4.3125H36.4688V1.47917H33.6875V4.3125H31.3542V6.33333H33.6875V11.8021C33.6875 13.0799 34.0417 14.0174 34.75 14.6146C35.4653 15.2049 36.5764 15.5 38.0833 15.5Z"
            fill="#423C66"
          />
          <path
            d="M46.7917 15.5625C47.5139 15.5625 48.1736 15.4653 48.7708 15.2708C49.375 15.0694 49.8993 14.7882 50.3438 14.4271C50.7951 14.066 51.1528 13.6424 51.4167 13.1563H51.4792V15.3125H54.25V4.3125H51.4792V9.78125C51.4792 10.3368 51.3854 10.8368 51.1979 11.2813C51.0174 11.7257 50.7604 12.1042 50.4271 12.4167C50.0938 12.7292 49.7014 12.9688 49.25 13.1354C48.7986 13.2951 48.2986 13.375 47.75 13.375C46.7014 13.375 45.9063 13.0868 45.3646 12.5104C44.8229 11.934 44.5521 11.1146 44.5521 10.0521V4.3125H41.7813V10.5625C41.7813 11.6181 41.9826 12.5208 42.3854 13.2708C42.7951 14.0139 43.3715 14.5833 44.1146 14.9792C44.8646 15.3681 45.7569 15.5625 46.7917 15.5625Z"
            fill="#423C66"
          />
          <path
            d="M56.4167 15.3125H59.1875V9.85417C59.1875 9.11111 59.3264 8.48611 59.6042 7.97917C59.8819 7.46528 60.2778 7.07986 60.7917 6.82292C61.3056 6.55903 61.9097 6.42708 62.6042 6.42708C62.8472 6.42708 63.0799 6.4375 63.3021 6.45833C63.5243 6.47917 63.7222 6.50694 63.8958 6.54167V4.19792C63.7639 4.17014 63.6042 4.14583 63.4167 4.125C63.2292 4.10417 63.0347 4.09375 62.8333 4.09375C61.9306 4.09375 61.1667 4.30903 60.5417 4.73958C59.9167 5.17014 59.4826 5.76736 59.2396 6.53125H59.1875V4.32292L56.4167 4.3125V15.3125Z"
            fill="#423C66"
          />
          <path
            d="M65.4688 15.4688C65.9201 15.4688 66.2951 15.3229 66.5938 15.0313C66.8993 14.7326 67.0521 14.3715 67.0521 13.9479C67.0521 13.5174 66.8993 13.1563 66.5938 12.8646C66.2951 12.566 65.9201 12.4167 65.4688 12.4167C65.0174 12.4167 64.6389 12.566 64.3333 12.8646C64.0347 13.1563 63.8854 13.5174 63.8854 13.9479C63.8854 14.3715 64.0347 14.7326 64.3333 15.0313C64.6389 15.3229 65.0174 15.4688 65.4688 15.4688Z"
            fill="#423C66"
          />
          <path
            d="M73.5729 15.5104C74.2882 15.5104 74.9514 15.4271 75.5625 15.2604C76.1736 15.0868 76.7118 14.8438 77.1771 14.5313C77.6424 14.2118 78.0035 13.8368 78.2604 13.4063H78.3125V15.3125H81.0833V8.30208C81.0833 7.42014 80.8472 6.66319 80.375 6.03125C79.9028 5.39931 79.2222 4.91319 78.3333 4.57292C77.4444 4.23264 76.375 4.0625 75.125 4.0625C73.9792 4.0625 72.9861 4.22222 72.1458 4.54167C71.3125 4.85417 70.6458 5.28819 70.1458 5.84375C69.6528 6.39236 69.3507 7.02083 69.2396 7.72917L69.2188 7.85417H71.8542L71.8854 7.76042C72.0243 7.25347 72.3646 6.85069 72.9063 6.55208C73.4549 6.25347 74.1806 6.10417 75.0833 6.10417C76.1181 6.10417 76.9132 6.30556 77.4688 6.70833C78.0313 7.11111 78.3125 7.67708 78.3125 8.40625V10.8229C78.3125 11.3785 78.1424 11.8646 77.8021 12.2813C77.4688 12.691 76.9965 13.0104 76.3854 13.2396C75.7813 13.4618 75.0729 13.5729 74.2604 13.5729C73.4201 13.5729 72.7431 13.441 72.2292 13.1771C71.7222 12.9063 71.4688 12.5 71.4688 11.9583V11.9375C71.4688 11.5139 71.6493 11.1771 72.0104 10.9271C72.3785 10.6771 72.9688 10.5417 73.7813 10.5208L79.7083 10.375V8.71875L73.3958 8.84375C72.2986 8.87153 71.4028 9.02083 70.7083 9.29167C70.0208 9.5625 69.5174 9.93403 69.1979 10.4063C68.8785 10.8785 68.7188 11.4306 68.7188 12.0625V12.0729C68.7188 12.816 68.9167 13.4444 69.3125 13.9583C69.7153 14.4653 70.2813 14.8507 71.0104 15.1146C71.7396 15.3785 72.5938 15.5104 73.5729 15.5104Z"
            fill="#423C66"
          />
          <path
            d="M83.3125 15.3125H86.0833V0.28125H83.3125V15.3125Z"
            fill="#423C66"
          />
          <path
            d="M88.4167 15.3125H91.1875V0.28125H88.4167V15.3125Z"
            fill="#423C66"
          />
          <path
            d="M95.5833 19C96.4931 19 97.2778 18.875 97.9375 18.625C98.5972 18.3819 99.1806 17.9792 99.6875 17.4167C100.194 16.8611 100.681 16.1146 101.146 15.1771L106.375 4.3125H103.458L99.4063 13.375L99.9375 12.9167H99.2083L99.7396 13.375L95.4792 4.3125H92.5208L97.9688 15.2396L97.6979 15.7188C97.4201 16.1701 97.0799 16.4826 96.6771 16.6563C96.2743 16.8368 95.7847 16.9271 95.2083 16.9271C94.9722 16.9271 94.7431 16.9132 94.5208 16.8854C94.2986 16.8576 94.0972 16.8229 93.9167 16.7813V18.8333C94.1389 18.8819 94.3924 18.9201 94.6771 18.9479C94.9688 18.9826 95.2708 19 95.5833 19Z"
            fill="#423C66"
          />
        </g>
      </svg>
    </div>
  </div>
  <div
    class="sc-beySPh ewWpC"
  >
    <div
      class="sc-hLQSwg fHrufj"
    >
      <svg
        fill="none"
        height="72"
        viewBox="0 0 72 72"
        width="72"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          id="GivingBlock_64"
        >
          <path
            d="M61.125 61.125V37.875V38.85V61.125ZM57 26.625C58.15 26.625 59.125 26.225 59.925 25.425C60.725 24.625 61.125 23.65 61.125 22.5C61.125 21.35 60.725 20.375 59.925 19.575C59.125 18.775 58.15 18.375 57 18.375C55.85 18.375 54.875 18.775 54.075 19.575C53.275 20.375 52.875 21.35 52.875 22.5C52.875 23.65 53.275 24.625 54.075 25.425C54.875 26.225 55.85 26.625 57 26.625ZM57 30C54.85 30 53.0625 29.2875 51.6375 27.8625C50.2125 26.4375 49.5 24.65 49.5 22.5C49.5 20.4 50.2125 18.625 51.6375 17.175C53.0625 15.725 54.85 15 57 15C59.1 15 60.875 15.725 62.325 17.175C63.775 18.625 64.5 20.4 64.5 22.5C64.5 24.65 63.775 26.4375 62.325 27.8625C60.875 29.2875 59.1 30 57 30ZM36 23.625C37.55 23.625 38.875 23.0875 39.975 22.0125C41.075 20.9375 41.625 19.6 41.625 18C41.625 16.45 41.075 15.125 39.975 14.025C38.875 12.925 37.55 12.375 36 12.375C34.4 12.375 33.0625 12.925 31.9875 14.025C30.9125 15.125 30.375 16.45 30.375 18C30.375 19.6 30.9125 20.9375 31.9875 22.0125C33.0625 23.0875 34.4 23.625 36 23.625ZM36 27C33.45 27 31.3125 26.1375 29.5875 24.4125C27.8625 22.6875 27 20.55 27 18C27 15.5 27.8625 13.375 29.5875 11.625C31.3125 9.875 33.45 9 36 9C38.5 9 40.625 9.875 42.375 11.625C44.125 13.375 45 15.5 45 18C45 20.55 44.125 22.6875 42.375 24.4125C40.625 26.1375 38.5 27 36 27ZM22.125 40.125C22.125 41.925 23.0125 43.9875 24.7875 46.3125C26.5625 48.6375 30.125 52.2 35.475 57L36 57.45L36.6 56.925C41.75 52.225 45.25 48.725 47.1 46.425C48.95 44.125 49.875 42.025 49.875 40.125C49.875 38.425 49.325 37 48.225 35.85C47.125 34.7 45.75 34.125 44.1 34.125C43.15 34.125 42.25 34.35 41.4 34.8C40.55 35.25 39.775 35.825 39.075 36.525L35.85 39.675H36.15L32.85 36.525C32.15 35.825 31.3875 35.25 30.5625 34.8C29.7375 34.35 28.85 34.125 27.9 34.125C26.25 34.125 24.875 34.7 23.775 35.85C22.675 37 22.125 38.425 22.125 40.125ZM18.675 40.125C18.675 37.675 19.6125 35.4875 21.4875 33.5625C23.3625 31.6375 25.5 30.675 27.9 30.675C29.35 30.675 30.7625 31.025 32.1375 31.725C33.5125 32.425 34.675 33.35 35.625 34.5L36 34.95L36.375 34.5C37.325 33.35 38.475 32.425 39.825 31.725C41.175 31.025 42.6 30.675 44.1 30.675C46.45 30.675 48.575 31.6375 50.475 33.5625C52.375 35.4875 53.325 37.675 53.325 40.125C53.325 42.625 52.3875 45.125 50.5125 47.625C48.6375 50.125 44.875 53.95 39.225 59.1L36 62.1L32.85 59.175C27.05 53.925 23.2375 50.0625 21.4125 47.5875C19.5875 45.1125 18.675 42.625 18.675 40.125ZM15 26.625C16.15 26.625 17.125 26.225 17.925 25.425C18.725 24.625 19.125 23.65 19.125 22.5C19.125 21.35 18.725 20.375 17.925 19.575C17.125 18.775 16.15 18.375 15 18.375C13.85 18.375 12.875 18.775 12.075 19.575C11.275 20.375 10.875 21.35 10.875 22.5C10.875 23.65 11.275 24.625 12.075 25.425C12.875 26.225 13.85 26.625 15 26.625ZM36.225 64.5V61.125H61.125V38.85C61.125 38.6 61.025 38.375 60.825 38.175C60.625 37.975 60.4 37.875 60.15 37.875H51.075V34.5H60.15C61.35 34.5 62.375 34.925 63.225 35.775C64.075 36.625 64.5 37.65 64.5 38.85V64.5H36.225ZM10.875 61.125H36.225V64.5H7.5V38.85C7.5 37.65 7.925 36.625 8.775 35.775C9.625 34.925 10.65 34.5 11.85 34.5H20.925V37.875H11.85C11.55 37.875 11.3125 37.975 11.1375 38.175C10.9625 38.375 10.875 38.6 10.875 38.85V61.125ZM15 30C12.85 30 11.0625 29.2875 9.6375 27.8625C8.2125 26.4375 7.5 24.65 7.5 22.5C7.5 20.4 8.2125 18.625 9.6375 17.175C11.0625 15.725 12.85 15 15 15C17.1 15 18.875 15.725 20.325 17.175C21.775 18.625 22.5 20.4 22.5 22.5C22.5 24.65 21.775 26.4375 20.325 27.8625C18.875 29.2875 17.1 30 15 30Z"
            fill="#423C66"
            id="Icon"
          />
        </g>
      </svg>
      <div>
        <h1>
          The giving block
        </h1>
        <p>
          Set up your donation goal!
        </p>
      </div>
      <div
        class="sc-eDLKkx kTUCbW"
      />
    </div>
    <form
      class="sc-guDLey kaoSRN"
    >
      <div
        class="sc-dmyCSP ewkgpU"
      >
        <label
          class="sc-kFCroH gFOUIW"
        >
          I can donate
        </label>
        <input
          class="sc-jTQCzO egtgQC"
          data-testid="input-amount"
          inputmode="decimal"
          placeholder="0.00"
          type="text"
          value=""
        />
      </div>
      <div
        class="sc-dmyCSP ewkgpU"
      >
        <label
          class="sc-kFCroH gFOUIW"
        >
          Every month until
        </label>
        <div
          class="sc-eTNRI gmKlKN"
          data-testid="month-picker"
        >
          <div
            class="sc-brSamD sc-ktwOfi iVtpkY kGhUds disabled"
            data-testid="prev-month"
          />
          <div
            class="sc-fmKFGs FlmDZ"
          >
            <div
              class="sc-fHejqy jQzMHP"
            >
              February
            </div>
            <div
              class="sc-blmEgr bKBkNc"
            >
              2024
            </div>
          </div>
          <div
            class="sc-brSamD sc-hABBmJ iVtpkY blQPhM"
            data-testid="next-month"
          />
        </div>
      </div>
      <div
        class="sc-gLLuof dHqmkN"
      >
        <div
          class="sc-iBdnpw bpqThd"
        >
          <div
            class="sc-fsYfdN jKWNBq"
          >
            Total amount
          </div>
          <div
            class="sc-qZrbh iGUJor"
          >
            $
            0.00
          </div>
        </div>
        <div
          class="sc-jsEeTM hCERCq"
        >
          You will be sending 
          <strong>
            $
            0.00
          </strong>
           
          every month, until
           
          <strong>
            February
             
            2024
          </strong>
          . Thank you!
        </div>
      </div>
      <div
        class="sc-irLvIq iawzZw"
      >
        <button
          class="sc-csKJxZ VlXTb"
          variant="white"
        >
          cancel
        </button>
        <button
          class="sc-csKJxZ eJLJRR"
          variant="primary"
        >
          continue
        </button>
      </div>
    </form>
  </div>
</div>
`;
